mhs/inbound/inbound/request/handler.py

```diff
@@ -101,24 +109,17 @@ def _extract_message_id(self, message):
         :return:
         """
         message_id = message.message_dictionary[ebxml_envelope.MESSAGE_ID]
-        if not message_id:
-            logger.info('008', "Didn't receive message id in inbound message")
-        else:
-            log.message_id.set(message_id)
-            logger.info('009', 'Found inbound message id on request.')
-        return message_id
+        log.inbound_message_id.set(message_id)
+        logger.info('009', 'Found inbound message id on request.')
 
     def extract_ref_message(self, message):
         """
         Extracts the reference-to message id and assigns it as the message Id in logging
         :param message:
         :return: the message id the inbound message is a response to
         """
-        try:
-            message_id = message.message_dictionary[ebxml_envelope.RECEIVED_MESSAGE_ID]
-            log.message_id.set(message_id)
-            logger.info('0008', 'Found message id on inbound message.')
-            return message_id
-        except KeyError:
-            logger.info('0007', "No Message reference found ")
-            return None
+        message_id = message.message_dictionary[ebxml_envelope.RECEIVED_MESSAGE_ID]
+        log.message_id.set(message_id)
```

Comment:
I'm not sure you're testing that any of these context variables are being set (see outbound handler tests for how I had to test this, I had to mock out the context variables as I think Tornado stops the event loop during the test, after you've called `self.fetch`.

Reply:
Should now be updated to check that they're set, as discussed the cases where these variables have been removed under the assumption they should always be there, and if they aren't the message parser will error first